builtin/credential/aws/path_login.go

```diff
@@ -208,13 +206,22 @@ func (b *backend) pathLoginUpdate(
 		return nil, err
 	}
 
+	clientNonce := data.Get("nonce").(string)
+	if clientNonce == "" && !storedIdentity.DisallowReauthentication {
```

Comment:
Unfortunately, I think that you need to not fail, because if reauthentication isn't allowed a nonce shouldn't be required. It would be strange to require one upon first login but not at reauthentication time. I realize this means that a lot of processing happens before the nonce is checked (except in validateMetadata), but...so be it. Chances are, since this is designed to be totally automated, this won't really be a massive problem for people since they should properly populate (or not) the nonce depending on their mode.